cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -432,14 +470,35 @@ angular.module(PKG.name + '.commons')
         }
 
         let connObj = {
-          source: conn.from,
           target: conn.to
         };
 
         if (conn.hasOwnProperty('condition')) {
           connObj.source = vm.instance.getEndpoints(`${conn.from}_condition_${conn.condition}`)[0];
         } else if (conn.hasOwnProperty('port')) {
           connObj.source = vm.instance.getEndpoint(`${conn.from}_port_${conn.port}`);
+        } else {
+          // if errortransform and alertpublisher target nodes then wait for vm.pluginsMap
+          if (targetNode.type === 'errortransform' || targetNode.type === 'alertpublisher') {
+            if (_.isEmpty(vm.pluginsMap)) {
+              let subAvailablePluginsForErrorAlertsConn = AvailablePluginsStore.subscribe(() => {
```

Comment:
This operation seems to be copy paste from line 338. Can we extract it out as a function and store refer the pluginsMap as a store in one place? Would be easier if we handle this in one place and unsubscribe once we have the plugin details.